src/ol/style/iconstyle.js

```diff
@@ -492,6 +516,7 @@ ol.style.IconImage_.prototype.handleImageLoad_ = function() {
   this.imageState_ = ol.style.ImageState.LOADED;
   this.size_ = [this.image_.width, this.image_.height];
   this.unlistenImage_();
+  this.replaceColor_();
```

Comment:
Applying the color will only work if the image is not tainted.
It should be moved after `determineTainting()`.